lib/components/form/advanced-settings-panel.tsx

```diff
@@ -0,0 +1,288 @@
+} from './util'
+import { setModeButtonEnabled } from './batch-settings'
+import { styledCheckboxCss } from './styled'
+import DateTimeModal from './date-time-modal'
+
+const PanelOverlay = styled.div`
+  height: 100%;
+  left: 0;
+  padding: 1.5em;
+  position: absolute;
+  top: 0;
+  width: 100%;
+  z-index: 100;
+  overflow-y: scroll;
+`
+
+const GlobalSettingsContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 10px;
+  margin-bottom: 2em;
+
+  ${styledCheckboxCss}
+`
+
+const CloseButton = styled.button`
+  background: transparent;
+  border: none;
+`
+
+const HeaderContainer = styled.div`
+  align-items: center;
+  display: flex;
+  justify-content: space-between;
+`
+
+const Subheader = styled.h2`
+  ${invisibleCss}
+`
+
+const PlanTripButton = styled.button`
+  align-items: center;
+  display: flex;
+  justify-content: center;
+  gap: 0.5em;
+  background-color: var(--main-base-color, ${blue[900]});
+  border: 0;
+  width: 47%;
+  height: 51px;
+  color: white;
+  font-weight: 700;
+`
+
+const ReturnToTripPlanButton = styled(PlanTripButton)`
+  border: 2px solid var(--main-base-color, ${blue[900]});
+  background-color: white;
+  color: var(--main-base-color, ${blue[900]});
+`
+const ButtonContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+  margin-top: 2em;
+`
+
+const DtSelectorContainer = styled.div`
+  padding: 1em;
+  margin: 2em 0;
+  background-color: ${blue[50]};
+
+  .date-time-modal {
+    padding: 0;
+
+    .main-panel {
+      margin: 0;
+
+      button {
+        padding: 6px 0;
+      }
+
+      .date-time-selector {
+        margin: 15px 0;
+      }
+    }
+  }
+`
+
+const AdvancedSettingsPanel = ({
+  closeAdvancedSettings,
+  currentQuery,
+  enabledModeButtons,
+  innerRef,
+  modeButtonOptions,
+  modeSettingDefinitions,
+  modeSettingValues,
+  onPlanTripClick,
+  routingQuery,
+  setQueryParam
+}: {
+  closeAdvancedSettings: () => void
+  currentQuery: any
+  enabledModeButtons: string[]
+  innerRef: RefObject<HTMLDivElement>
+  modeButtonOptions: ModeButtonDefinition[]
+  modeSettingDefinitions: ModeSetting[]
+  modeSettingValues: ModeSettingValues
+  onPlanTripClick: () => void
+  routingQuery: () => void
+  setQueryParam: (evt: any) => void
+}): JSX.Element => {
+  const baseColor = getBaseColor()
+  const accentColor = baseColor || blue[900]
+
+  const intl = useIntl()
+  const closeButtonText = intl.formatMessage({
+    id: 'components.BatchSearchScreen.closeAdvancedPreferences'
+  })
+  const headerText = intl.formatMessage({
+    id: 'components.BatchSearchScreen.advancedHeader'
+  })
+
+  // @ts-expect-error Context not typed
+  const { ModeIcon } = useContext(ComponentContext)
+
+  const planTrip = () => {
+    alertUserTripPlan(intl, currentQuery, onPlanTripClick, routingQuery)
+    closeAdvancedSettings()
+  }
+
+  const closeAndAnnounce = () => {
+    closeAdvancedSettings()
+    toast.success(intl.formatMessage({ id: 'actions.user.preferencesSaved' }))
```

Comment:
The toast is very attention-grabbing and could be avoided by clearer design

Reply:
I'm open to alternatives! There needs to be some sort of user feedback that the changes have been applied when you return to the home page, and this toast is in line with how we show that feedback in other places across the app.

Reply:
The animation can be used as a form of feedback. See https://bootcamp.uxdesign.cc/toast-or-no-toast-tips-on-providing-feedback-in-product-design-210d24a7e547 or https://www.nngroup.com/articles/ten-usability-heuristics/ #1, #8 and #4. There are few products that have a toast confirmation when saving a simple preferences pane